src/main.js

```diff
@@ -1,144 +1,175 @@
 'use babel'
 
-// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
-import { CompositeDisposable } from 'atom'
-import { hasValidScope } from './validate/editor'
-import {
-  atomConfig,
-  jobConfig,
-  getMigrations,
-  subscribe as configSubscribe
-} from './atom-config'
-
-// Internal variables
-const idleCallbacks = new Set()
-
 // Dependencies
-// NOTE: We are not directly requiring these in order to reduce the time it
-// takes to require this file as that causes delays in Atom loading this package
-let path
-let worker
-let configInspector
-let debug
-let linterMessage
-let knownRules
-
-const loadDeps = () => {
-  if (!path) {
-    path = require('path')
-  }
-  if (!worker) {
-    worker = require('./worker-manager')
-  }
-  if (!configInspector) {
-    configInspector = require('./eslint-config-inspector')
-  }
-  if (!debug) {
-    debug = require('./debug')
-  }
-  if (!linterMessage) {
-    linterMessage = require('./linter-message')
-  }
-  if (!knownRules) {
-    knownRules = require('./rules').default
-  }
-}
+// NOTE: Requiring files adds significant time to package startup. To avoid this
+// delay, we are *not* requiring at the top of this file. Use idleCallbacks to
+// attempt to pre-cache requires for fast resolution if scheduler allows. Since
+// all requires are statically defined and inside some function, Atom should be
+// able "snapshot" the package.
+//
+const dependencies = [
+  () => require('./validate/editor'),
+  () => require('./linter-message'),
+  () => require('./worker-manager'),
+  () => require('./eslint-config-inspector'),
+  () => require('./debug'),
+  () => require('./rules'),
+  () => require('path'),
+  () => require('./atom-config'),
+]
+
+// Miscellaneous idle tasks.
+//
+const idleTasks = [
+  // Pre-start the worker
+  () => require('./worker-manager').task.start(),
+  // Install peer packages
+  () => require('atom-package-deps').install('linter-eslint'),
+]
+
+// Run idle tasks with all reasonable attempts to avoid blocking.
+//
+const makeIdleRunner = () => {
+  const { requestIdleCallback, cancelIdleCallback } = window
+  let id
+  let tasks
+
+  const loadLazily = (deadline) => {
```

Comment:
This isn't _just_ being used to lazily load dependencies. Since this is the name that will show in CPU profiles and we want some way of identifying it how about `linterEslintWork`?

Reply:
That's a good point. When I came up with that name, it was only being used for deps, so the name was pretty clear. But I do think something on the lines of "lazy" or "idle" is important to expressing intention.... Um. Maybe just `runLazily`? Not sure.

Reply:
The fact that it is running in an idle callback is enough for my concerns, no matter what `linterEslint` needs to be in the name of something _pretty_ high up the stack trace so you can trace back where this is coming from.